components/dashboard/src/teams/TeamSettings.tsx

```diff
@@ -110,37 +116,47 @@ export default function TeamSettings() {
     return (
         <>
             <OrgSettingsPage>
-                <Heading2>Organization Name</Heading2>
-                <Subheading className="max-w-2xl">
-                    This is your organization's visible name within Gitpod. For example, the name of your company.
-                </Subheading>
-                {errorMessage && (
+                <Heading2>Organization Details</Heading2>
+                <Subheading className="max-w-2xl">Details of your organization within Gitpod.</Subheading>
+
+                {updateOrg.isError && (
                     <Alert type="error" closable={true} className="mb-2 max-w-xl rounded-md">
-                        {errorMessage}
+                        <span>Failed to update organization information: </span>
+                        <span>{updateOrg.error.message || "unknown error"}</span>
                     </Alert>
                 )}
                 {updated && (
                     <Alert type="message" closable={true} className="mb-2 max-w-xl rounded-md">
                         Organization name has been updated.
                     </Alert>
                 )}
-                <div className="flex flex-col lg:flex-row">
-                    <div>
-                        <div className="mt-4 mb-3">
-                            <h4>Name</h4>
-                            <input type="text" value={teamName} onChange={onNameChange} />
-                        </div>
-                    </div>
-                </div>
-                <div className="flex flex-row">
-                    <button
-                        className="primary"
-                        disabled={org?.name === teamName || !!errorMessage}
-                        onClick={updateTeamInformation}
+                <form onSubmit={updateTeamInformation}>
+                    <TextInputField
+                        label="Name"
+                        hint="The name of your company or organization"
+                        value={teamName}
+                        error={teamNameError.message}
+                        onChange={setTeamName}
+                        onBlur={teamNameError.onBlur}
+                    />
+
+                    <TextInputField
+                        label="Slug"
+                        hint="The slug will be used for easier signin and discovery"
+                        value={slug}
+                        error={slugError.message}
+                        onChange={setSlug}
+                        onBlur={slugError.onBlur}
+                    />
```

Comment:
**issue**: Using the same slug twice does not throw an error—we could probably not allow same slugs within an instance, right?

Reply:
Thanks for catching this!